README.md

```diff
@@ -1 +1,146 @@
+
+### Usage
+
+#### Run Command
+
+You will have to configure a config.json to configure your Amazon Dash Buttons and give information to services. 
+
+To start the docker container use the following command: 
+
+```bash
+docker run -d --name dash-button --net=host --restart unless-stopped --privileged -v /pathToConfig.json:/config.json fusonic/amazon-dash-buttons:latest
+```
+
+Change pathToConfig.json to match your config.json file.
+
+##### Flags needed due to:
+
+* `--net` - have access to the network, which dash-buttons are connected to
+* `--privileged` - being able to inspect packages in the host system
+* `-v` - mount the json file into container
+
+#### Create config.json
+
+There is a example.config.json located in the root of the project, where you can see how your config file could look like.
+
+Here is an example on how to configure a dash button:
+
+```json 
+"buttons": [
+  {
+    "mac": "xx:xx:xx:xx:xx:xx",
+    "provider": "slack",
+    "providerConfig": {
+    "stuffForProvider": "beep"
+    }
+  }
+]
+```
+
+And a provider:
+
+```json
+"providerConfig": {
+  "slack": {
+    "stuffForProvider": "beep"
+  }
+}
+```
+
+A complete config.json could look like this:
+
+```json
+{
+  "buttons": [
+    {
+      "mac": "xxxx",
+      "provider": "slack",
+      "providerConfig": {
+        "channel": "xxxx",
+        "username": "dash-bot",
+        "text": ":grin:",
+        "icon_emoji": "true"
+      }
+    }
+  ],
+  "providerConfig": {
+    "slack": {
+      "token": "xxxx"
+    }
+  }
+}
+```
+#### Providers
+Currently the following providers are supported:
+* [Bring](https://getbring.com/)
+* [Slack](https://slack.com/)
+
+If the provider you are looking for is currently not supported, feel free to integrate it into the project. Therefore have a look at the Contribute section.
+
+#### Documentation on Providers
+
+There is a detailed README.md file covering how to configure your config.json in the files/opt/dash-button/scripts/providers/ folder.
+
+You can also find the file here well formated on [Github](https://github.com/fusonic/amazon-dash-buttons/blob/master/files/opt/dash-button/scripts/providers/README.md).
+### Contribute
+#### Add a new Provider
+To add a new provider you must create a JavaScript file in files/opt/dash-button/scripts/providers/providerName.js and register the provider as a module.
+
+```javascript
+module.exports = {
+    dashHandler : function (sender, providerConfig) {
+       // magic
+    }
+};
+
+```
+
+Note: the file must have the same name as the provider declared in the config.json!
+
+The `dashHandler` function is called with two parameters:
+
+ * `sender` - containing information from button
+ * `providerConfig` - containing information 
+ 
+ With these parameters you can access information declared in the config.json file.
+ 
+#### Run command for developing
+We created a rundev file that mounts the scrips folder as a volume, which is useful for developing new features.
```

Comment:
Please re-check formatting of markdown documents and make sure that each headline has a blank line before and after (except the one on the first line). Makes reading it in plaintext much easier.